### ISSUE TITLE
SEEK doesn't work correctly when there are multiple noField() arguments and ORDER BY is uniform and Settings.renderRowConditionForSeekClause is true

### ISSUE THREAD

**Issue by wojcikt** [jOOQ/jOOQ#16450]
### Expected behavior

When all fields are of `NoField` instance, no conditions should be generated.

### Actual behavior

Some conditions are generated.

### Steps to reproduce the problem

```java
import org.jooq.SQLDialect;
import org.jooq.impl.DSL;
import org.jooq.impl.SQLDataType;

	DSL.using(SQLDialect.POSTGRES)
			.select(DSL.asterisk())
			.from(DSL.name("t"))
			.orderBy(
					DSL.field("x", SQLDataType.INTEGER),
					DSL.field("y", SQLDataType.INTEGER)
			)
			.seek(
					DSL.noField(SQLDataType.INTEGER),
					DSL.noField(SQLDataType.INTEGER)
			)
```

gives us:

```sql
select *
from "t"
where y > null
order by
  x,
  y
```

This line seems to be the problem: https://github.com/jOOQ/jOOQ/blob/7e22b9818e785c2c5fc199de34476d3ac2890dae/jOOQ/src/main/java/org/jooq/impl/SelectQueryImpl.java#L4453

### jOOQ Version

jOOQ Pro 3.18.6

### Database product and version

PostgreSQL 15.5 (Debian 15.5-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit

### Java Version

openjdk version "17.0.1" 2021-10-19

**Comment by lukaseder** [comment]
Thank you very much for your report. I don't recall implementing `noField()` semantics for the `SEEK` clause, but obviously, this should be done.

I'll look into this soon.

**Comment by lukaseder** [comment]
> I don't recall implementing `noField()` semantics for the `SEEK` clause, but obviously, this should be done.

I recalled wrong. This has been fixed in 3.18.0 and 3.17.7:

- https://github.com/jOOQ/jOOQ/issues/14395

But you're right, the loop is incorrect as it doesn't adjust the loop variable while removing things.

**Comment by lukaseder** [comment]
Fixed in versions:

- 3.20.0
- 3.19.7 (https://github.com/jOOQ/jOOQ/issues/16462)
- 3.18.14 (https://github.com/jOOQ/jOOQ/issues/16463)
- 3.17.23 (https://github.com/jOOQ/jOOQ/issues/16464)
